src/formatters/functions.rs

```diff
@@ -1158,7 +1165,8 @@ pub fn format_function_declaration(
         shape
     )
     .update_leading_trivia(FormatTriviaType::Append(leading_trivia));
-    let formatted_function_name = format_function_name(ctx, function_declaration.name(), shape);
+    let formatted_function_name = format_function_name(ctx, function_declaration.name(), shape)
+        .update_trailing_trivia(FormatTriviaType::Append(function_definition_trivia));
 
     let shape = shape + (9 + strip_trivia(&formatted_function_name).to_string().len()); // 9 = "function "
```

Comment:
A point worth mentioning is that our "shape" calculation here (and potentially elsewhere) will not take into account the fact that a space was added after the function name (since we explicitly strip the trivia afterwards). Shape is used to store the amount of line budget used so far and how many characters we have left on a line. This opens up a very small edge case where the definition line now hits `column_width + 1` characters because of the space, but shape only sees `column_width` and therefore we don't reformat correctly. It's quite a minor edge case though, unsure if it is worth rectifying until someone brings it up

Reply:
I'd be happy to fix the edge case if I had any idea where/how, but I wasn't actually that clear on the internals of how that was being done or where extra info should be passed to get it back. Given what you say here I'm guessing the same edge case might exist for some other scenarios unrelated to this PR, no?